geth/rpc/call_raw_test.go

```diff
@@ -9,23 +9,23 @@ import (
 )
 
 func TestNewSuccessResponse(t *testing.T) {
-	res := []byte(`"3434=done"`)
-	got := newSuccessResponse(res, nil)
-
-	expected := `{"jsonrpc":"2.0","id":0,"result":"3434=done"}`
-	require.Equal(t, expected, got)
-
-	res = []byte(`{"field": "value"}`)
-	got = newSuccessResponse(res, nil)
-
-	expected = `{"jsonrpc":"2.0","id":0,"result":{"field":"value"}}`
-	require.Equal(t, expected, got)
-
-	res = []byte(`{"field": "value"}`)
-	got = newSuccessResponse(res, json.RawMessage(`42`))
+	cases := []struct {
+		name     string
+		result   json.RawMessage
+		id       json.RawMessage
+		expected string
+	}{
+		{"string", json.RawMessage(`"3434=done"`), nil, `{"jsonrpc":"2.0","id":0,"result":"3434=done"}`},
+		{"struct_nil_id", json.RawMessage(`{"field": "value"}`), nil, `{"jsonrpc":"2.0","id":0,"result":{"field":"value"}}`},
+		{"struct_non_nil_id", json.RawMessage(`{"field": "value"}`), json.RawMessage(`42`), `{"jsonrpc":"2.0","id":42,"result":{"field":"value"}}`},
+	}
 
-	expected = `{"jsonrpc":"2.0","id":42,"result":{"field":"value"}}`
-	require.Equal(t, expected, got)
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			got := newSuccessResponse(test.result, test.id)
+			require.Equal(t, test.expected, got)
+		})
+	}
 }
```

Comment:
By the way, why this way? If you wrote them with "duplicated" code, the number of lines would be the same but they'd be more readable. It's just these are tests, they better be written in a way that is easy to read and modify, no need to generalise logic. What do you think?

Reply:
@tiabc any new test case now will take +1 line now vs +5 duplicated lines. Any change into testing (let's say, we're switching to GoConvey at somepoint would be also lot easier. Plus less probability of shadowing issues, etc.